3_summarize/src/do_annual_thermal_metric_tasks.R

```diff
@@ -111,6 +173,20 @@ combine_thermal_metrics <- function(target_name, ...) {
   purrr::map(list(...), function(ind) readRDS(as_data_file(ind))) %>% purrr::reduce(bind_rows) %>% readr::write_csv(target_name)
 }
 
+combine_model_thermal_metrics <- function(out_ind, model_id_colname, ...) {
+  data_file <- as_data_file(out_ind)
+  purrr::map(list(...), function(ind) {
+    rds_file <- as_data_file(ind)
+    file_pattern <- "(?<modelid>.*)_(?<sitenum>nhdhr_.*)_annual_thermal_metrics.rds" # based on target_name of `calc_annual_metrics` step
+    readRDS(rds_file) %>% 
+      mutate(model_id = str_match(basename(rds_file), file_pattern)[2]) %>% 
```

Comment:
Not necessary for a change, but wondering if `tidry::extract()` is more appropriate in these situations compared to `mutate(.. = str_match)`. 

But I'm not sure what `basename(rds_file)` looks like so maybe I am on the wrong track with this comment. Again, not a necessary change since I think these two options would do the same thing, just perhaps cleaner for `extract`.

Reply:
I always forget about `extract()`! I am only wanting the second group, not a column per group, so I think I will leave in this case.